Frameworks/CoreGraphics/CGGeometry.mm

```diff
@@ -87,11 +87,85 @@ bool CGRectMakeWithDictionaryRepresentation(CFDictionaryRef dict, CGRect* rect)
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 void CGRectDivide(CGRect rect, CGRect* slice, CGRect* remainder, CGFloat amount, CGRectEdge edge) {
-    UNIMPLEMENTED();
+    RETURN_IF(slice == nullptr);
+    RETURN_IF(remainder == nullptr);
+
+    amount = (amount < 0) ? 0 : amount;
+    rect = CGRectStandardize(rect);
+
+    /*
+    * (0,0)                      (width,0)
+    * x-----------------------------x
+    * |                             |
+    * |                             |
+    * |                             |
+    * |                             |
+    * |                             |
+    * x-----------------------------x
+    * (0,height)               (width,height)
+    *
+    *
+    * The division is based on CGRectEdge, which edge to divide.
+    * e.g CGRectMinYEdge
+    *
+    *   (0,0)                      (width,0)
+    *    x-----------------------------x
+    *    | slice            | (amount) |
+    *    |                  v          |
+    * (0,amount)------------------(width,amount)
+    *    | reminder                    |
+    *    |                             |
+    *    x-----------------------------x
+    *   0,height-amount)               (width,height-amount)
+    */
+
+    // Set both to rect, then update the sizes.
+    *slice = rect;
+    *remainder = rect;
+
+    switch (edge) {
+        case CGRectMinYEdge:
+            remainder->origin.y += amount;
+            if (amount >= rect.size.height) {
+                remainder->size.height = 0;
```

Comment:
This would return a weird value for invalid rects. What does the reference platform return for these cases?